express_relay/sdk/python/searcher/examples/simple_searcher.py

```diff
@@ -0,0 +1,160 @@
+import argparse
+import asyncio
+import logging
+
+from searcher_utils import BidInfo, SearcherClient
+
+from schema.openapi_client.models.opportunity_bid import OpportunityBid
+from schema.openapi_client.models.opportunity_params_with_metadata import OpportunityParamsWithMetadata
+
+logger = logging.getLogger(__name__)
+
+VALID_UNTIL = 1_000_000_000_000
+
+class SimpleSearcher(SearcherClient):
```

Comment:
You could also instantiate a new SearcherClient object within SimpleSearcher. However, SimpleSearcher uses a lot of the functionality of SearcherClient, so it seems like a wrapper around the basic client functionality (interact with the server). Some of the logic is wrapped, e.g. the ws path calls a function in SearcherClient that then routes to the opportunity handler function defined in SimpleSearcher, which includes a call to submit_bid in SearcherClient. This makes it more natural to express SimpleSearcher as an extension of the client

Reply:
The fact that it uses a lot of the functionalities does not mean we should subclass it here. If we were about to extend its functionality or change some behavior, this would make sense. But now we are putting the logic of searcher (bidding, assessing opportunities) in the same place that we put the logic of connecting to the server while these are very separate concerns.